### PR TITLE
Add tests to init liskAPI - Closes #314

### LINKED ISSUE #314: Add tests for lisk api initialization
Part of #261 

Very few of the default properties of lisk api are tested currently.

### PR DESCRIPTION
**Pull request by Tosch110**
Closes #314

### CODE REVIEW

File: src/api/privateApi.js [hunk 1]
```diff
@@ -67,61 +67,61 @@ export function getURLPrefix() {
...
 	}
 
-	const randomIndex = Math.floor((Math.random() * peers.length));
-	return peers[randomIndex];
+	const randomIndex = Math.floor((Math.random() * nodes.length));
+	return nodes[randomIndex];
 }
 
 /**
  * @method selectNode
```
Comment (willclarktech) [inline src/api/privateApi.js]: Can we give this function a more illuminating name? `setNode`, `refreshNode`, `selectNewNode`?

File: src/api/privateApi.js [hunk 2]
```diff
@@ -67,61 +67,61 @@ export function getURLPrefix() {
...
- * @return peer string
+ * @return node string
  * @private
  */
 
 export function selectNode() {
```
Comment (willclarktech) [inline src/api/privateApi.js]: Should it take a node as an optional argument?

File: src/api/privateApi.js [hunk 3]
```diff
@@ -130,8 +130,8 @@ export function selectNode() {
  */
 
 export function banNode() {
```
Comment (willclarktech) [inline src/api/privateApi.js]: Should this also take a node as an optional argument?

File: src/api/privateApi.js [hunk 4]
```diff
@@ -142,29 +142,29 @@ export function banNode() {
  */
 
 export function checkReDial() {
```
Comment (willclarktech) [inline src/api/privateApi.js]: Can we rename this?

File: src/api/privateApi.js [hunk 5]
```diff
@@ -142,29 +142,29 @@ export function banNode() {
...
-	const peers = getPeers.call(this);
+	const nodes = getNodes.call(this);
 
-	// RandomPeer discovery explicitly set
-	if (this.randomPeer === true) {
+	// randomNode discovery explicitly set
```
Comment (willclarktech) [inline src/api/privateApi.js]: Do we still need all these comments?

File: test/api/liskApi.js [hunk 6]
```diff
@@ -99,25 +100,113 @@ describe('Lisk API module', () => {
...
+				LSK = new LiskAPI({ ssl: true });
+				(LSK).should.have.property('ssl').be.true();
+			});
+
+			it('should set SSL to false on initialization when passed as an option', () => {
+				LSK = new LiskAPI({ ssl: false });
+				(LSK).should.have.property('ssl').be.false();
+			});
+
+			it('should set SSL to false on initialization when no options is passed', () => {
```
Comment (willclarktech) [inline test/api/liskApi.js]: `when no SSL option is passed`

File: test/api/liskApi.js [hunk 7]
```diff
@@ -99,25 +100,113 @@ describe('Lisk API module', () => {
...
+				LSK = new LiskAPI({ });
+				(LSK).should.have.property('ssl').be.false();
+			});
+		});
+
+		describe('randomNode', () => {
+			it('should set randomNode to true when no node is given and randomNode not explicitly set', () => {
+				(LSK).should.have.property('randomNode').be.true();
```
Comment (willclarktech) [inline test/api/liskApi.js]: We should consistently test with `new LiskAPI()` or `new LiskAPI({})` (or both).

File: test/api/liskApi.js [hunk 8]
```diff
@@ -99,25 +100,113 @@ describe('Lisk API module', () => {
...
+
+			it('should set randomNode to true on initialization when passed as an option', () => {
+				LSK = new LiskAPI({ randomNode: true });
+				(LSK).should.have.property('randomNode').be.true();
+			});
+
+			it('should set randomNode to false on initialization when passed as an option', () => {
+				LSK = new LiskAPI({ randomNode: false });
+				(LSK).should.have.property('randomNode').be.false();
+			});
```
Comment (willclarktech) [inline test/api/liskApi.js]: What about when `randomNode` is not explicitly set but a node is provided?

File: test/api/liskApi.js [hunk 9]
```diff
@@ -99,25 +100,113 @@ describe('Lisk API module', () => {
...
+
+			it('should set testnet true and port to 100 on initialization when passed as an option', () => {
+				LSK = new LiskAPI({ port: 100, testnet: true });
+				(LSK).should.have.property('port').be.equal(100);
+			});
+		});
+
+		describe('nodes', () => {
+			it('should set all nodes list to provided nodes on initialization when passed as an option', () => {
```
Comment (willclarktech) [inline test/api/liskApi.js]: `nodes lists`

File: test/api/privateApi.js [hunk 10]
```diff
@@ -172,90 +172,90 @@ describe('privateApi module', () => {
...
 		const { selectNode } = privateApi;
-		const customNode = 'customPeer';
-		const getRandomPeerResult = externalNode;
+		const customNode = 'customNode';
+		const getRandomNodeResult = externalNode;
 
-		let getRandomPeerStub;
-		let restoreGetRandomPeer;
+		let getRandomNodeStub;
+		let restoregetRandomNode;
```
Comment (willclarktech) [inline test/api/privateApi.js]: `restoreGetRandomNode`

File: src/api/liskApi.js [hunk 11]
```diff
@@ -98,15 +100,15 @@ export default class LiskAPI {
 	}
 
 	/**
-	 * @method getPeers
+	 * @method getNodes
 	 * @return {object}
 	 */
 
-	getPeers() {
+	getNodes() {
```
[the remark this answers is not in the document]
Comment (Tosch110) [reply src/api/liskApi.js]: Probably. I am not sure about this function, it was explicitly written by lisk-nano for lisk-nano

File: src/api/privateApi.js (hunk 2 above)
Comment (Tosch110) [reply src/api/privateApi.js]: No, it is not `setNode` but `selectNewNode`. It returns a new node depending on the given options.

File: test/api/liskApi.js (hunk 7 above)
Comment (Tosch110) [reply test/api/liskApi.js]: `new LiskAPI({})` is part of the beforeEach hook (now with curly braces)

File: test/api/liskApi.js (hunk 8 above)
Comment (Tosch110) [reply test/api/liskApi.js]: Thanks, the description was probably misleading. Either randomNode is passed as `true|false` or not, defaulting to `true`. A passed node has no influence on it

Comment (willclarktech) [reply test/api/liskApi.js]: But in case a node is provided, we want `randomNode` to default to `false`, no?

File: src/api/privateApi.js [hunk 12]
```diff
@@ -136,25 +136,15 @@ export function banActiveNode() {
...
-			} else if (this.options.nethash === netHashOptions.call(this).mainnet.nethash) {
-				this.setTestnet(false);
-				return true;
-			}
-			return false;
-		}
+	if (this.randomNode) {
 		return nodes.some(node => !this.bannedNodes.includes(node));
 	}
 	return false;
```
Comment (willclarktech) [inline src/api/privateApi.js]: Prefer ternary syntax:
```js
return this.randomNode
    ? nodes.some(node => !this.bannedNodes.includes(node))
    : false;
```

File: src/api/privateApi.js [hunk 13]
```diff
@@ -61,111 +61,92 @@ export function getURLPrefix() {
...
+ * @return  {String}
  * @private
  */
 
-export function getRandomPeer() {
-	const peers = getPeers.call(this)
-		.filter(peer => !this.bannedPeers.includes(peer));
+export function getRandomNode() {
+	const nodes = getNodes.call(this)
+		.filter(node => !this.bannedNodes.includes(node));
```
Comment (willclarktech) [inline src/api/privateApi.js]: We check against `this.bannedNodes.includes(node)` a lot. A new `privateApi` method `isBanned` would make these checks more DRY.

**Pull request merged**